src/core/Value.cc

```diff
@@ -389,7 +393,12 @@ class tostring_visitor : public boost::static_visitor<std::string>
   std::string operator()(const VectorType& v) const {
     // Create a single stream and pass reference to it for list elements for optimization.
     std::ostringstream stream;
-    (tostream_visitor(stream))(v);
+    try {
+      (tostream_visitor(stream))(v);
+    } catch (EvaluationException& e) {
+      LOG(message_group::Error, Location::NONE, "", "Stack exhausted while trying to convert a vector to EchoString");
```

Comment:
I would just use `e.what()` here for the string, so that the message from the exception constructor is not needlessly duplicated / unused.